data/meta/ShipDef.lua

```diff
@@ -0,0 +1,44 @@
+-- Copyright © 2008-2024 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+-- This file implements type information about C++ modules for Lua static analysis
+
+---@meta
+
+---@alias ShipDef.Thrust { UP:number, DOWN:number, LEFT:number, RIGHT:number, FORWARD:number, REVERSE:number }
+
+---@class ShipDef
+---@field id string
+---@field name string
+---@field shipClass string
+---@field manufacturer string
+---@field modelName string
+---@field cockpitName string
+---@field tag ShipTypeTag
+---@field roles string[]
+--
+---@field angularThrust number
+---@field linearThrust ShipDef.Thrust
+---@field linAccelerationCap ShipDef.Thrust
+---@field effectiveExhaustVelocity number
+---@field thrusterFuelUse number -- deprecated
```

Comment:
Q: why add already-deprecated fields to a new feature? Or is this to support legacy functionality?

Reply:
This was deprecated many, many, many years ago and has never been removed. I'm actually not sure if it's still in use, this is only showing as "new" as it's documenting existing code/structures for the Lua language server we use.